### PR TITLE
GUAC-1126 Add caching of GET requests in REST API services.

### PR DESCRIPTION
**Pull request by jmuehlner**

### CODE REVIEW

File: guacamole/src/main/webapp/app/rest/services/cacheService.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+         * @type $cacheFactory.Cache
+         */
+        protocols : $cacheFactory('API-PROTOCOLS'),
+
+        /**
+         * Shared cache used by both userService and permissionService.
+         *
+         * @type $cacheFactory.Cache
+         */
+        users : $cacheFactory('API-USER')
```
Comment (mike-jumper) [inline guacamole/src/main/webapp/app/rest/services/cacheService.js]: ```API-USERS```?

Comment (jmuehlner) [inline guacamole/src/main/webapp/app/rest/services/cacheService.js]: I agree

**Comment by mike-jumper** [comment]
Looks good, but what happens when someone logs out? (or in)

When the user logs out, or a new user logs in, they should be given clear caches.

**Comment by mike-jumper** [comment]
Should be GUAC-1126.

**Pull request closed without merging**